From patchwork Fri Jan 15 17:37:41 2021
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: John Levon <john.levon@nutanix.com>
X-Patchwork-Id: 12023787
Return-Path: <kvm-owner@kernel.org>
X-Spam-Checker-Version: SpamAssassin 3.4.0 (2014-02-07) on
	aws-us-west-2-korg-lkml-1.web.codeaurora.org
X-Spam-Level: 
X-Spam-Status: No, score=-16.8 required=3.0 tests=BAYES_00,
	HEADER_FROM_DIFFERENT_DOMAINS,INCLUDES_CR_TRAILER,INCLUDES_PATCH,
	MAILING_LIST_MULTI,SPF_HELO_NONE,SPF_PASS,URIBL_BLOCKED,USER_AGENT_GIT
	autolearn=ham autolearn_force=no version=3.4.0
Received: from mail.kernel.org (mail.kernel.org [198.145.29.99])
	by smtp.lore.kernel.org (Postfix) with ESMTP id 3B789C433DB
	for <kvm@archiver.kernel.org>; Fri, 15 Jan 2021 18:22:20 +0000 (UTC)
Received: from vger.kernel.org (vger.kernel.org [23.128.96.18])
	by mail.kernel.org (Postfix) with ESMTP id 13E7D23A58
	for <kvm@archiver.kernel.org>; Fri, 15 Jan 2021 18:22:20 +0000 (UTC)
Received: (majordomo@vger.kernel.org) by vger.kernel.org via listexpand
        id S1733084AbhAOSWL (ORCPT <rfc822;kvm@archiver.kernel.org>);
        Fri, 15 Jan 2021 13:22:11 -0500
Received: from 88-98-93-30.dsl.in-addr.zen.co.uk ([88.98.93.30]:34960 "EHLO
        sent" rhost-flags-OK-FAIL-OK-FAIL) by vger.kernel.org with ESMTP
        id S1733294AbhAOSWJ (ORCPT <rfc822;kvm@vger.kernel.org>);
        Fri, 15 Jan 2021 13:22:09 -0500
X-Greylist: delayed 2553 seconds by postgrey-1.27 at vger.kernel.org;
 Fri, 15 Jan 2021 13:22:09 EST
Received: from jlevon by sent with local (Exim 4.93)
        (envelope-from <john.levon@nutanix.com>)
        id 1l0T3g-00B1tx-IL; Fri, 15 Jan 2021 17:38:48 +0000
From: John Levon <john.levon@nutanix.com>
To: levon@movementarian.org, virtualization@lists.linux-foundation.org,
        kvm@vger.kernel.org, jasowang@redhat.com, mst@redhat.com
Cc: John Levon <john.levon@nutanix.com>
Subject: [PATCH] use pr_warn_ratelimited() for vq_err()
Date: Fri, 15 Jan 2021 17:37:41 +0000
Message-Id: <20210115173741.2628737-1-john.levon@nutanix.com>
X-Mailer: git-send-email 2.25.1
MIME-Version: 1.0
Precedence: bulk
List-ID: <kvm.vger.kernel.org>
X-Mailing-List: kvm@vger.kernel.org

vq_err() is used to report various failure states in vhost code, but by
default uses pr_debug(), and as a result doesn't record anything unless
enabled via dynamic debug. We'll change this so we get something recorded
in the log in these failure cases. Guest VMs (and userspace) can trigger
some of these messages, so we want to use the pr_warn_ratelimited()
variant.

Signed-off-by: John Levon <john.levon@nutanix.com>
---
 drivers/vhost/vhost.h | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/drivers/vhost/vhost.h b/drivers/vhost/vhost.h
index b063324c7669..cb4ef78c84ba 100644
--- a/drivers/vhost/vhost.h
+++ b/drivers/vhost/vhost.h
@@ -228,10 +228,10 @@ int vhost_init_device_iotlb(struct vhost_dev *d, bool enabled);
 void vhost_iotlb_map_free(struct vhost_iotlb *iotlb,
 			  struct vhost_iotlb_map *map);
 
-#define vq_err(vq, fmt, ...) do {                                  \
-		pr_debug(pr_fmt(fmt), ##__VA_ARGS__);       \
-		if ((vq)->error_ctx)                               \
-				eventfd_signal((vq)->error_ctx, 1);\
+#define vq_err(vq, fmt, ...) do {                                \
+		pr_warn_ratelimited(pr_fmt(fmt), ##__VA_ARGS__); \
+		if ((vq)->error_ctx)                             \
+			eventfd_signal((vq)->error_ctx, 1);      \
 	} while (0)
 
 enum {
